examples/armbench-segmentation/armbench_segmentation/__init__.py

```diff
@@ -0,0 +1,4 @@
+CACHE_DICTS = {'loss': {},
```

Comment:
let's try to reduce the memory of this cache (this is the reason for the OOM)

- use hashing instead of str(tensors)
- caching only the last batch for metrics that the visualizers don't use (for example the loss). also consider using the decorator `lru_cache(max_size=1)` for that
-  I think it would also be fine to cache the last sample for the visualizer metrics, we would get most of the speedup and reduce significantly the memory.

I think it can take 1-2 days, so maybe it's better to limit the number of samples in the script and do it when we have time.  @yotamazriel